sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/Storage.java

```diff
@@ -934,7 +933,7 @@ public void run() {
                     eventListener.onDataStoreOperationResult(
                             pendingOperation.getOperation(),
                             null,
-                            new DocumentError(e));
```

Comment:
If it's already a storage exception we should not wrap it so need to test exception type before.

Reply:
Fixed here and in the `Page` constructor

Reply:
> Task :sdk:appcenter-storage:testDebugUnitTest

com.microsoft.appcenter.storage.NetworkStateChangeStorageTest > pendingDeleteOperationWithCosmosDb409Error FAILED
    java.lang.AssertionError at NetworkStateChangeStorageTest.java:341

com.microsoft.appcenter.storage.NetworkStateChangeStorageTest > pendingDeleteOperationDeletesExpiredOperationOnCosmosDb500Error FAILED
    java.lang.AssertionError at NetworkStateChangeStorageTest.java:341

com.microsoft.appcenter.storage.NetworkStateChangeStorageTest > pendingDeleteOperationWithCosmosDb500Error FAILED
    java.lang.AssertionError at NetworkStateChangeStorageTest.java:188